### PR TITLE
Add agents and groups management methods

### PR DESCRIPTION
**Pull request by kogierman**

### CODE REVIEW

File: configuration/api.go [hunk 1]
```diff
@@ -114,6 +114,75 @@ func (a *API) GetBot(id string) (*BotAgentDetails, error) {
...
+	return resp, err
+}
+
+// ListAgents returns all Agents within a license.
+func (a *API) ListAgents(groupIDs []int32, fields []string) ([]*Agent, error) {
+	var resp listAgentsResponse
+	err := a.Call("list_agents", &listAgentsRequest{
+		Filters: AgentsFilters{
+			GroupIDs: groupIDs,
+		},
```
Comment (Adimus11) [inline configuration/api.go:142]: We should not add `filters.group_ids` if provided group_ids from client are empty or empty array

File: configuration/structures.go [hunk 2]
```diff
@@ -118,3 +118,46 @@ type Group struct {
...
+	AvatarPath         string        `json:"avatar_path,omitempty"`
+	JobTitle           string        `json:"job_title,omitempty"`
+	Mobile             string        `json:"mobile,omitempty"`
+	MaxChatsCount      uint          `json:"max_chats_count,omitempty"`
+	AwaitingApproval   bool          `json:"awaiting_approval,omitempty"`
+	Groups             []GroupConfig `json:"groups,omitempty"`
+	WorkScheduler      WorkScheduler `json:"work_scheduler,omitempty"`
+	Notifications      []string      `json:"notifications,omitempty"`
+	EmailSubscriptions []string      `json:"email_subscriptions,omitempty"`
+}
```
Comment (Adimus11) [inline configuration/structures.go:136]: I'm wondering if you should add support to all other fields not listed in docs 🤔

Comment (kogierman) [reply configuration/structures.go:136]: if it's not listed in the docs, then it's not in the code :D SDK is public, docs are public, they should match each other

or do we have some undocumented public fields?

Comment (Adimus11) [reply configuration/structures.go:136]: We have a few of them, but I think that this most used and available to users are listed in docs.

File: configuration/api.go [hunk 3]
```diff
@@ -114,6 +114,75 @@ func (a *API) GetBot(id string) (*BotAgentDetails, error) {
...
 
+// CreateAgent creates a new Agent with specified parameters within a license
+func (a *API) CreateAgent(agent *Agent) (string, error) {
+	var resp createAgentResponse
+	err := a.Call("create_agent", agent, &resp)
+
+	return resp.ID, err
+}
+
+// GetAgent returns the info about an Agent specified by id
```
Comment (litwinow) [inline configuration/api.go:125]:
```suggestion
// GetAgent returns the info about an Agent specified by it's id (ie. login)
```

File: configuration/api.go [hunk 4]
```diff
@@ -114,6 +114,75 @@ func (a *API) GetBot(id string) (*BotAgentDetails, error) {
...
+	var resp getAgentResponse
+	err := a.Call("get_agent", &getAgentRequest{
+		ID:     id,
+		Fields: fields,
+	}, &resp)
+
+	return resp, err
+}
+
+// ListAgents returns all Agents within a license.
```
Comment (litwinow) [inline configuration/api.go:136]: please unify dot at end of the comments across whole document (dot should be used everywhere)

File: configuration/api.go [hunk 5]
```diff
@@ -114,6 +114,75 @@ func (a *API) GetBot(id string) (*BotAgentDetails, error) {
...
+		Filters: AgentsFilters{
+			GroupIDs: groupIDs,
+		},
+		Fields: fields,
+	}, &resp)
+	return resp, err
+}
+
+// UpdateAgent updates the properties of an Agent specified by id.
+func (a *API) UpdateAgent(agent *Agent) error {
```
Comment (litwinow) [inline configuration/api.go:149]: I think `id` should be separate param and `Agent` shouldn't contain it

File: configuration/structures.go (hunk 2 above)
Comment (litwinow) [reply configuration/structures.go:136]: non-documented fields should not be listed here

File: configuration/api.go [hunk 6]
```diff
@@ -50,7 +50,7 @@ func (a *API) UnregisterWebhook(id string) error {
 }
 
 // CreateBot allows to create bot and returns its ID
-func (a *API) CreateBot(name, avatar string, status BotStatus, maxChats uint, defaultPriority GroupPriority, groups []*BotGroupConfig, webhooks *BotWebhooks) (string, error) {
+func (a *API) CreateBot(name, avatar string, status BotStatus, maxChats uint, defaultPriority GroupPriority, groups []*GroupConfig, webhooks *BotWebhooks) (string, error) {
```
Comment (litwinow) [inline configuration/api.go:53]: Can you please split all these functions to separate files, grouped as in docs? (ie. `agents.go`, `bots.go`, `groups.go`, `properties.go` and `webhooks.go`)

**Pull request merged**